cpp/src/arrow/builder.cc

```diff
@@ -56,10 +76,46 @@ Status ArrayBuilder::Advance(int32_t elements) {
 
 Status ArrayBuilder::Reserve(int32_t elements) {
   if (length_ + elements > capacity_) {
+    // TODO(emkornfield) power of 2 growth is potentially suboptimal
```

Comment:
Aside: should we do 1.5x growth everywhere (this is the folly approach IIRC -- is there more research on this subject?)